src/SOS/SOS.Extensions/DebuggerServices.cs

```diff
@@ -31,10 +31,23 @@ internal enum OperatingSystem
 
         private readonly HostType _hostType;
 
+        /// <summary>
+        /// A pointer to the underlying IDebugClient interface if the host is DbgEng.
+        /// </summary>
+        public IDebugClient DebugClient { get; }
+
         internal DebuggerServices(IntPtr punk, HostType hostType)
             : base(new RefCountedFreeLibrary(IntPtr.Zero), IID_IDebuggerServices, punk)
         {
             _hostType = hostType;
+
+            // This uses COM marshalling code, so we also check that the OSPlatform is Windows.
+            if (hostType == HostType.DbgEng && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                object obj = Marshal.GetObjectForIUnknown(punk);
+                if (obj is IDebugClient client)
+                    DebugClient = client;
```

Comment:
Does this need to be released in HostServices.Uninitialize()? It is only important for the rare unload SOS/reload SOS scenario to check the debug client's ref count from leaking.

Reply:
It won't leak, it will be released in the finalizer.

Do you explicitly dispose DebuggingServices?  If you do, I can add an explicit Marshal.FinalReleaseComObject in the Dispose method.  If you don't explicitly dispose DebuggingServices, then it will be disposed at the same speed as DebuggingServices does.

Reply:
DebuggerServices is released in Uninitialize() [here]( https://github.com/dotnet/diagnostics/blob/789b69fb7a4258d88ca48aea7f34789c0aa5de57/src/SOS/SOS.Extensions/HostServices.cs#L415) but is not disposable.  It sounds like nothing needs to be done here.